src/main/java/org/truffleruby/core/time/RubyDateFormatter.java

```diff
@@ -655,18 +655,21 @@ public static Rope formatToRopeBuilderFast(Token[] compiledPattern, ZonedDateTim
 
                 case FORMAT_NANOSEC: {
                     final int nano = dt.getNano();
-                    final LazyIntRope nanoRope = new LazyIntRope(nano);
+                    final LazyIntRope microSecondRope = new LazyIntRope(nano / 1000);
 
                     final int length = 6;
-                    final int padding = length - nanoRope.characterLength();
+                    final int padding = length - microSecondRope.characterLength();
 
                     if (padding == 0) {
-                        appendRope = nanoRope;
+                        appendRope = microSecondRope;
                     } else if (padding < 0) {
-                        appendRope = substringNode.executeSubstring(nanoRope, 0, length);
+                        appendRope = substringNode.executeSubstring(microSecondRope, 0, length);
                     } else {
                         appendRope = concatNode
```

Comment:
Is this case even possible actually?
I would think not, because the maximum value is 999_999_999 nanos, / 100 so 999_999 and that can never be bigger than 6 digits.